research/object_detection/utils/variables_helper.py

```diff
@@ -151,5 +156,24 @@ def get_variables_available_in_checkpoint(variables,
       logging.warning('Variable [%s] is not available in checkpoint',
                       variable_name)
   if isinstance(variables, list):
-    return vars_in_ckpt.values()
+    return list(vars_in_ckpt.values())
   return vars_in_ckpt
+
+
+def get_global_variables_safely():
+  """If not executing eagerly, returns tf.global_variables().
+
+  Raises a ValueError if eager execution is enabled,
+  because the variables are not tracked when executing eagerly.
+
+  If executing eagerly, use a Keras model's .variables property instead.
+
+  Returns:
+    The result of tf.global_variables()
+  """
+  with tf.init_scope():
```

Comment:
Hi

This particular new line is breaking backward compatibility.

I'm following [this documentation](https://tensorflow-object-detection-api-tutorial.readthedocs.io/en/latest/training.html) which [force to use  TensorFlow 1.9](https://tensorflow-object-detection-api-tutorial.readthedocs.io/en/latest/install.html#install-tensorflow-cpu-for-python). However, accordin to [this](issue) `init_scope` is available on version 1.1 or more.

Can you suggest a workarround for this line?

Thanks

Reply:
the document you linked is not official is quite outdated. Our latest code requires tf 1.12.